src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1658,76 +1657,51 @@ private void MakeSeparatorList(ReadOnlySpan<char> separators, ref ValueListBuild
 
         private void MakeSeparatorListVectorized(ref ValueListBuilder<int> sepListBuilder, char c, char c2, char c3)
         {
-            // Redundant test so we won't prejit remainder of this method
-            // on platforms without SSE.
-            if (!Sse41.IsSupported)
+            if (!Vector128.IsHardwareAccelerated)
             {
                 throw new PlatformNotSupportedException();
             }
 
-            // Constant that allows for the truncation of 16-bit (FFFF/0000) values within a register to 4-bit (F/0)
-            Vector128<byte> shuffleConstant = Vector128.Create(0x00, 0x02, 0x04, 0x06, 0x08, 0x0A, 0x0C, 0x0E, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF);
+            nuint offset = 0;
+            nuint lengthToExamine = (nuint)(uint)Length;
+
+            ref ushort source = ref Unsafe.As<char, ushort>(ref _firstChar);
 
             Vector128<ushort> v1 = Vector128.Create((ushort)c);
             Vector128<ushort> v2 = Vector128.Create((ushort)c2);
             Vector128<ushort> v3 = Vector128.Create((ushort)c3);
 
-            ref char c0 = ref MemoryMarshal.GetReference(this.AsSpan());
-            int cond = Length & -Vector128<ushort>.Count;
-            int i = 0;
-
-            for (; i < cond; i += Vector128<ushort>.Count)
+            while (offset <= lengthToExamine - (nuint)Vector128<ushort>.Count)
```

Comment:
Above is L1618 we guard by `Vector128<ushort>.Count * 2`, so when coming to here we know that there for sure enough elements available. This this check isn't need at this point. So you could change the loop to da `do-while` loop. Thus the first iteration is without any (further) pre-condition, and after the iteration the check for more available elements is done.

Reply:
Thanks. I'll add a Debug.Assert on entry to make it a little more obvious to the reader that its a precondition